### PR TITLE
Action: Github Get Open Branches

### PR DESCRIPTION
**Pull request by shloka-bhalgat-unskript**
### [Issue-255](https://github.com/unskript/Awesome-CloudOps-Automation/issues/255)

Added a new action to get open branches for a user in a repository. (Limit-100)

### Testing
Screenshot 2023-02-27 at 4 34 45 PM

### Checklist:
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository.

### CODE REVIEW

File: Github/legos/github_get_open_branches/github_get_open_branches.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        :type repository: string
+        :param repository: Full name of the GitHub repository. Eg: "unskript/Awesome-CloudOps-Automation"
+
+        :rtype: List of branches for a user for a repository
+    """
+    result = []
+    try:
+        user = handle.get_user(login=owner)
+        repos = user.get_repos()
+        repo_name = owner+"/"+repository
```
Comment (dougsillars) [inline Github/legos/github_get_open_branches/github_get_open_branches.py:47]: if repository is the full  name (unskript/Awesome-CloudOps-Automation)
does this one make it unskript/unskript/Awesome-CloudOps-Automation?

**Review by dougsillars — CHANGES_REQUESTED** [review]
i think repository should not have the owner/ - but just the name of the repo.  Maybe just change the comment to make that clear.

**Review by dougsillars — APPROVED** [review]
Thanks!  LGTM

File: Github/legos/github_get_open_branches/github_get_open_branches.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            if repo.full_name == repo_name:
+                branches = repo.get_branches()
+                [result.append(branch.name) for branch in branches[:100]]
+    except GithubException as e:
+        if e.status == 403:
+            return [f"You need push access"]
+        elif e.status == 404:
+            return [f"No such username or repository"]
+        else:
+            return [e.data]
```
Comment (harshal-netserv) [inline Github/legos/github_get_open_branches/github_get_open_branches.py:60]: @shloka-bhalgat-unskript we should raise the exception instead of return.

**Review by harshal-netserv — APPROVED** [review]

**Pull request merged**